lua/lazyvim/plugins/extras/lang/typescript-vscode.lua

```diff
@@ -0,0 +1,263 @@
+            end,
+            desc = "Open Vtsls Log",
+          },
+          {
+            "<leader>cR",
+            function()
+              require("vtsls").commands.rename_file(0)
+            end,
+            desc = "Rename File",
+          },
+          {
+            "<leader>cu",
+            function()
+              require("vtsls").commands.file_references(0)
+            end,
+            desc = "Show File Uses(References)",
+          },
+          {
+            "<leader>ctc",
+            function()
+              require("vtsls").commands.goto_project_config(0)
+            end,
+            desc = "Open Project Config",
+          },
+        },
+        settings = {
+          vtsls = {
+            experimental = {
+              completion = {
+                enableServerSideFuzzyMatch = true,
+              },
+            },
+          },
+          javascript = {
+            format = {
+              indentSize = vim.o.shiftwidth,
+              convertTabsToSpaces = vim.o.expandtab,
+              tabSize = vim.o.tabstop,
+            },
+            -- enables inline hints
+            inlayHints = {
+              parameterNames = { enabled = "literals" },
+              parameterTypes = { enabled = true },
+              variableTypes = { enabled = true },
+              propertyDeclarationTypes = { enabled = true },
+              functionLikeReturnTypes = { enabled = true },
+              enumMemberValues = { enabled = true },
+            },
+            -- otherwise it would ask every time if you want to update imports, which is a bit annoying
+            updateImportsOnFileMove = {
+              enabled = "always",
+            },
+            -- cool feature, but increases ram usage
+            -- referencesCodeLens = {
+            --   enabled = true,
+            --   showOnAllFunctions = true,
+            -- },
+          },
+          typescript = {
+            format = {
+              indentSize = vim.o.shiftwidth,
+              convertTabsToSpaces = vim.o.expandtab,
+              tabSize = vim.o.tabstop,
+            },
+            updateImportsOnFileMove = {
+              enabled = "always",
+            },
+            inlayHints = {
+              parameterNames = { enabled = "literals" },
+              parameterTypes = { enabled = true },
+              variableTypes = { enabled = true },
+              propertyDeclarationTypes = { enabled = true },
+              functionLikeReturnTypes = { enabled = true },
+              enumMemberValues = { enabled = true },
+            },
+            -- enables project wide error reporting similar to vscode
+            -- tsserver = {
+            --   experimental = {
+            --     enableProjectDiagnostics = true,
+            --   },
+            -- },
+          },
+        },
+      }
+    end,
+    dependencies = {
+      {
+        "yioneko/nvim-vtsls",
+        handlers = {},
+      },
+    },
+  },
```

Comment:
```suggestion
  },
    {
    "folke/which-key.nvim",
    opts = {
      defaults = {
        ["<leader>ct"] = { name = "+Project", ["🚫"] = "which_key_ignore" },
        ["<leader>cL"] = { name = "+Log", ["🚫"] = "which_key_ignore" },
      },
    },
  },
```
If you would like, you can include that to show prefixes in `which-key` for the 2 corresponding keymaps.